exercises/concept/black-jack/.docs/introduction.md

```diff
@@ -1,185 +1,249 @@
+If all code points in both strings match and are _**in the same order**_, the two strings are considered equal.
+This comparison is done in a 'pair-wise' fashion - first-to-first, second-to-second, etc.
+Unlike in Python 2.x, in Python 3.x, `str` and `bytes` cannot be directly coerced/compared.
+
 
 ```python
->>> 3 <= 1
+>>> 'Python' > 'Rust'
 False
->>> 2.99 <= 3
-True
->>> 1 <= 1
+
+>>> 'Python' > 'JavaScript'
 True
->>> 0 <= 1
+
+# Examples with Mandarin.
+# hello < goodbye
+>>> '你好' < '再见'  
 True
-```
 
-## Not equal to
+# ord() of first characters
+>>> ord('你'), ord('再')
+(20320, 20877)
 
-Operator `!=` tests if the first operand's value is not equal to the second one's
+# ord() of second characters
+>>> ord('好'), ord('见')
+(22909, 35265)
 
-```python
->>> 3 != 1
-True
->>> 2.99 != 3
-True
->>> 1 != 1
+# And with Korean words.
+# Pretty < beautiful.
+>>> '예쁜' < '아름다운'  
 False
->>> 0 != 1
-True
+
+>>> ord('예'), ord('아')
+(50696, 50500)
 ```
 
-## Identity test
+## Comparison Chaining
+
+Comparison operators can be chained _arbitrarily_ -- meaning that they can be used in any combination of any length.
+Note that the evaluation of an expression takes place from `left` to `right`.
+
+
+As an example, `x < y <= z` is equivalent to `x < y` `and` `y <= z`, except that `y` is evaluated **only once**.
+In both cases, `z` is _not_ evaluated **at all** when `x < y` is found to be `False`.
+This is often called `short-circuit evaluation` - the evaluation stops if the truth value of the expression has already been determined.
+
+`Short circuiting` is supported by various boolean operators, functions, and also by comparison chaining in Python.
+Unlike many other programming languages, including `C`, `C++`, `C#`, and `Java`, chained expressions like `a < b < c` in Python have a conventional [mathematical interpretation][three way boolean comparison] and precedence.
 
-Operator `is` tests if the first and second operand is the same object.
 
 ```python
-# comparing non-object type `is` will raise warning
->>> 1 is 1
-<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
+>>> x = 2
+>>> y = 5
+>>> z = 10
+
+>>> x < y < z
 True
->>> 1 is 2
-<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
+
+>>> x < y > z
 False
->>> x = int(4)
->>> y = x
->>> x is y
-True
->>> y = int(5)
->>> x is y
+
+>>> x > y < z
 False
 ```
 
-Operator `is not` tests if the first and second operand is not the same object.
+
+## Comparing object identity
+
+The operators `is` and `is not` test for object [_identity_][object identity], as opposed to object _value_.
+An object's identity never changes after creation and can be found by using the [`id()`][id function] function.
+
+\<apple\> `is` \<orange\> evaluates to `True` if _**and only if**_ `id(<apple>)` == `id(<orange>)`.
+apple `is not` orange yields the inverse.
```

Comment:
I think this should be 

```markdown
`apple is not orange` yields the inverse.
```

instead.

Reply:
yeah..or whatever we end up deciding.  Although it should be `<apple> is not <orange>` -- if we want to keep to the `<>` signifying a variable or object.

Reply:
If that `<>` indicates variable or object in Python speak then that is what should be used.  I think in the context, perhaps it is clear to focus on the `is not` rather than the variable name used (and is why we use foo and bar as well known "can be whatever, as allowed by the syntax of the language in that case sensitive interpretation".